spec/controllers/api/v3/analytics/user_analytics_controller_spec.rb

```diff
@@ -25,87 +25,123 @@
       end
     end
 
-    context 'html' do
-      render_views
+    context 'when diabetes management is enabled' do
+      before :each do
+        request.env['HTTP_X_USER_ID'] = request_user.id
+        request.env['HTTP_X_FACILITY_ID'] = request_facility.id
+        request.env['HTTP_AUTHORIZATION'] = "Bearer #{request_user.access_token}"
+      end
 
-      describe 'facility has data' do
-        it 'gets html when requested' do
-          get :show, format: :html
+      context 'html' do
+        render_views
 
-          expect(response.status).to eq(200)
-          expect(response.content_type).to eq('text/html')
-        end
 
-        it 'has the sync nudge card' do
-          get :show, format: :html
+        describe 'facility has data' do
+          it 'gets html when requested' do
+            get :show, format: :html
 
-          expect(response.body).to match(/Tap "Sync" on the home screen for new data/)
-        end
+            expect(response.status).to eq(200)
+            expect(response.content_type).to eq('text/html')
+          end
 
-        it 'has the registrations card' do
-          get :show, format: :html
+          it 'has the sync nudge card' do
+            get :show, format: :html
 
-          expect(response.body).to match(/Registered/)
-        end
+            expect(response.body).to match(/Tap "Sync" on the home screen for new data/)
+          end
 
-        it 'has the follow-ups card' do
-          get :show, format: :html
+          it 'has the registrations card' do
+            get :show, format: :html
 
-          expect(response.body).to match(/Follow-up hypertension patients/)
-        end
+            expect(response.body).to match(/Registered/)
+          end
 
-        it 'has the hypertension control card' do
-          get :show, format: :html
+          it 'has the follow-ups card' do
+            get :show, format: :html
 
-          expect(response.body).to match(/Hypertension control/)
-        end
+            expect(response.body).to match(/Follow-up patients/)
+          end
```

Comment:
You can combine all of these expectations into one spec. Since we are rendering views here, these specs are going to be really slow. Its definitely worth it to combine simple assertions like this to one spec for real heavy things like this.